From 43e11a3b8001ff8f7de32db42c99cd6ced5da891 Mon Sep 17 00:00:00 2001
From: John Linn <john.linn@xilinx.com>
Date: Tue, 21 Jun 2011 10:27:34 -0600
Subject: [PATCH 160/628] Xilinx: ARM: Device Cfg: Fixed driver compile error

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 8564526b2656a607518a7ad39d37fd7138f6639d

A minor annoyance, the last patch was not correct.

Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/char/xilinx_devcfg.c |   20 ++++++++++----------
 1 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/drivers/char/xilinx_devcfg.c b/drivers/char/xilinx_devcfg.c
index 776014e..9e357fb 100644
--- a/drivers/char/xilinx_devcfg.c
+++ b/drivers/char/xilinx_devcfg.c
@@ -1462,6 +1462,16 @@ static int __devexit xdevcfg_drv_remove(struct platform_device *pdev)
 	return 0;		/* Success */
 }
 
+#ifdef CONFIG_OF
+static struct of_device_id xdevcfg_of_match[] __devinitdata = {
+	{ .compatible = "xlnx,xdevcfg", },
+	{ /* end of table */}
+};
+MODULE_DEVICE_TABLE(of, xdevcfg_of_match);
+#else
+#define xdevcfg_of_match NULL
+#endif /* CONFIG_OF */
+
 /* Driver Structure */
 static struct platform_driver xdevcfg_platform_driver = {
 	.probe = xdevcfg_drv_probe,
@@ -1473,16 +1483,6 @@ static struct platform_driver xdevcfg_platform_driver = {
 	},
 };
 
-#ifdef CONFIG_OF
-static struct of_device_id xdevcfg_of_match[] __devinitdata = {
-	{ .compatible = "xlnx,xdevcfg", },
-	{ /* end of table */}
-};
-MODULE_DEVICE_TABLE(of, xdevcfg_of_match);
-#else
-#define xdevcfg_of_match NULL
-#endif /* CONFIG_OF */
-
 /**
  * xdevcfg_module_init -  register the Device Configuration.
  *
-- 
1.7.5.4

